Source/CarthageKit/Project.swift

```diff
@@ -145,6 +144,23 @@ public final class Project { // swiftlint:disable:this type_body_length
 		let cartfileURL = directoryURL.appendingPathComponent(Constants.Project.cartfilePath, isDirectory: false)
 		let privateCartfileURL = directoryURL.appendingPathComponent(Constants.Project.privateCartfilePath, isDirectory: false)
 
+		// Because the file system Carthage is interacting with may be case sensitive a validation must be
+		// performed to ensure proper file naming. See: https://github.com/Carthage/Carthage/issues/1336
+		func isValidProjectFiles() -> Bool {
+			let workingDirectory = (try? FileManager.default.contentsOfDirectory(at: directoryURL, includingPropertiesForKeys: [], options: .skipsHiddenFiles)) ?? []
+
+			func isValidFile(name: String) -> Bool {
+				let cartfileURL = directoryURL.appendingPathComponent(name)
+				if FileManager.default.fileExists(atPath: cartfileURL.path) {
+					return workingDirectory.filter({ $0.lastPathComponent == name }).count == 1
+				}
+				return true
+			}
+			// DISCUSS: This could cause some issues. Based on the name `Constants.Project.cartfilePath` it suggests that
+			// the string is a path and not a file name.
```

Comment:
I think it's fine to rename the constants to specify that they're filenames.

Reply:
All the cartfilePath constants contain docs specifying that they are relative paths to the project. If this is truly intent I would hate to change it to a file name. From searching the project the only thing I could see being a problem later is if the location of the Cartfile is no longer at project root. I suppose we could deal with that if it ever came up. 

I only want to make sure the intention is clear for what the constant is supposed to represent.

Reply:
Yeah, good looking out, we could comment and use `assert` (which gets compiled out in Release)`.

```swift
[Constants.Project.cartfilePath, Constants.Project.privateCartfilePath]
	.some_function { filename in // relative path used here as file-system resource name
		assert(filename.pathComponents.count == 1)
		…
```

Reply:
I should ask as well, `Constants.Project.resolvedCartfilePath` should be checked too, right?

Reply:
Not sure `Constants.Project.resolvedCartfilePath` needs checked since Carthage manages that file. Unless we want to validate that the file has not been tampered with. But I think this scenario would be exposed if `carthage bootstrap` was ran after the file was tampered with.

So, I guess we would need to understand the workflow and at what points this could occur. But again, this scenario means that a user is going in and deliberately messing with the `resolvedCartfilePath` file. And in that case we shouldn't try and solve for people being stupid. Just my opinion...

Reply:
> But again, this scenario means that a user is going in and deliberately messing with the resolvedCartfilePath file. And in that case we shouldn't try and solve for people being stupid.

My concern is more bad tooling/scripts messing with filename case — even smart people frequently fail to vet those comprehensively.

Reply:
I am fine with adding in the check for the resolved file. But does it makes sense within the workflow to have the resolved file checked at `loadCombinedCartfiles`?